### PR TITLE
Adding side_effect

### PR DESCRIPTION
**Pull request by erez-work**

### CODE REVIEW

**Comment by erez-work** [comment]
@davidstrouk FYI

File: gamla/functional_test.py [hunk 1]
```diff
@@ -707,3 +707,23 @@ def test_concat_with():
...
+
+
+def test_side_effect():
+    side_result = []
+    side_effect = functional_generic.compose_left(
+        functional.multiply(2),
+        side_result.append,
+    )
+    assert functional_generic.side_effect(side_effect)(2) == 2
+    assert side_result[0] == 4
```
Comment (uriva) [inline gamla/functional_test.py:719]: consider `side_result == [4]`

File: gamla/functional_test.py [hunk 2]
```diff
@@ -707,3 +707,23 @@ def test_concat_with():
...
+
+
+async def test_side_effect_async():
+    side_result = []
+    side_effect = functional_generic.compose_left(
+        functional.currying.curry(_equals)(2),
```
Comment (uriva) [inline gamla/functional_test.py:725]: import `currying` directly

File: gamla/functional_test.py [hunk 3]
```diff
@@ -707,3 +707,23 @@ def test_concat_with():
...
+        side_result.append,
+    )
+    assert await functional_generic.side_effect(side_effect)(2) == 2
+    assert side_result[0]
```
Comment (uriva) [inline gamla/functional_test.py:729]: `side_result == [True]`

**Review by uriva — APPROVED** [review]

**Pull request merged**